Sources/AWSLambdaRuntimeCore/LambdaRunner.swift

```diff
@@ -28,6 +33,7 @@ extension Lambda {
         init(eventLoop: EventLoop, configuration: Configuration) {
             self.eventLoop = eventLoop
             self.runtimeClient = RuntimeClient(eventLoop: self.eventLoop, configuration: configuration.runtimeEngine)
+            self.tracer = XRayRecorder(eventLoopGroupProvider: .shared(eventLoop))
```

Comment:
I see... wanting to share the EL can be good... We'll need to figure out how to allow this when using the abstract API. @tomerd I guess users don't really create the EL, the framework creates it right?

Since the tracing API can't depend on NIO, we can't solve it by initializers in there; we could solve it perhaps by lambda offering some `bootstrapTracer: EL -> TracingInstrument`? Or users have to create the event loop (group), then bootstrap their tracer, then run the lambda... a bit more ceremony I guess hm. Need to look into this more

Reply:
> I see... wanting to share the EL can be good... We'll need to figure out how to allow this when using the abstract API. @tomerd I guess users don't really create the EL, the framework creates it right?

YES

> Since the tracing API can't depend on NIO, we can't solve it by initializers in there; we could solve it perhaps by lambda offering some bootstrapTracer: EL -> TracingInstrument?

+1

> Or users have to create the event loop (group), then bootstrap their tracer, then run the lambda... a bit more ceremony I guess hm.

-1

Reply:
@fabianfett I would really appreciate any suggestions how/where to configure/bootstrap tracer.

The challenge here is to pass the event loop which is currently not exposed by `InstrumentationSystem` not Lambda runtime.

`XRayRecorder` can create its own but its not optimal (low tier lambda configuration has just a fraction of vCPU, starting from 1/8 as far as I remember).

```swift
let instrument = XRayRecorder() // same as XRayRecorder(eventLoopGroupProvider: .createNew)
defer { instrument.shutdown() }
InstrumentationSystem.bootstrap(instrument)
```

this will most likely be affected (get easier?) by `ServiceLifecycle` integration https://github.com/swift-server/swift-aws-lambda-runtime/pull/141